yarl/_url.py

```diff
@@ -1123,7 +1126,59 @@ def join(self, url):
         # See docs for urllib.parse.urljoin
         if not isinstance(url, URL):
             raise TypeError("url should be URL")
-        return URL(urljoin(str(self), str(url)), encoded=True)
+        other: URL = url
+        scheme = other.scheme or self.scheme
+        parts = {
+            k[len("raw_") :]: getattr(self, k) or ""
+            for k in ("raw_authority", "raw_path", "raw_query_string", "raw_fragment")
```

Comment:
Instead of slicing up `k` it might be a bit faster to unpack a tuple of the known value

Reply:
You are right, the compiler does not bother to optimize.

```python
>>> def x():
...     return "ab"[len("a"):]
... 
>>> import dis
>>> dis.dis(x)
  2           0 LOAD_CONST               1 ('ab')
              2 LOAD_GLOBAL              0 (len)
              4 LOAD_CONST               2 ('a')
              6 CALL_FUNCTION            1
              8 LOAD_CONST               0 (None)
             10 BUILD_SLICE              2
             12 BINARY_SUBSCR
             14 RETURN_VALUE
```